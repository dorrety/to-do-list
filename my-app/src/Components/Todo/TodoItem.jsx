import { useContext } from "react";
import { Context } from "../../context";

export default function TodoItem({ text, id, completed }) {
    const { toggleTodo, deleteTodo } = useContext(Context)

    return (
        <label>
            <li className="todo-item">
                <input
                    className="todo-checkbox"
                    type="checkbox" 
                    checked={completed}
                    onChange={() => toggleTodo(id)}
                    />
                <span className={`todo-text ${completed ? 'checked' : ''}`}>{text}</span>
                <button className="delete-button" onClick={() => deleteTodo(id)}>-</button>
            </li>
        </label>
    )
}