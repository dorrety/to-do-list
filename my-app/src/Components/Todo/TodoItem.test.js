import { render } from '@testing-library/react';
import TodoItem from './TodoItem';
import { Context } from '../../context';

describe('TodoItem', () => {
    let contextValue;

    beforeEach(() => {
        contextValue = {
            toggleTodo: jest.fn(),
            deleteTodo: jest.fn(),
        };
    });

    it('checked text todo', () => {
        const { getByText  } = render(
        <Context.Provider value={contextValue}>
          <TodoItem text="Test todo" id={1} completed={false} />
        </Context.Provider>
        );

        expect(getByText(/Test Todo/i)).toMatchSnapshot();
    })

    it('checkbox is clicked', () => {
        const { getByRole } = render(
        <Context.Provider value={contextValue}>
            <TodoItem text="Test Todo" id={1} completed={false} />
        </Context.Provider>
        );

        const checkbox = getByRole('checkbox');
        checkbox.click();

        expect(contextValue.toggleTodo).toHaveBeenCalledWith(1);
    });

    it('delete button', () => {
        const { getByRole } = render(
        <Context.Provider value={contextValue}>
            <TodoItem text="Test Todo" id={1} completed={false} />
        </Context.Provider>
        );

        const deleteButton = getByRole('button');
        deleteButton.click();

        expect(contextValue.deleteTodo).toHaveBeenCalledWith(1);
    });
});