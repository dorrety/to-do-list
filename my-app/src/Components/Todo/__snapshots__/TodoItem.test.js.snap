// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem checked text todo 1`] = `
<span
  class="todo-text "
>
  Test todo
</span>
`;
